modules/vault/src/main/java/org/apache/ignite/internal/vault/persistence/PersistentVaultService.java

```diff
@@ -112,59 +93,44 @@ public void start() {
         }
     }
 
-    /** {@inheritDoc} */
     @Override
     public void close() {
-        IgniteUtils.shutdownAndAwaitTermination(threadPool, 10, TimeUnit.SECONDS);
-
-        futureTracker.cancelInFlightFutures();
-
-        RocksUtils.closeAll(options, db);
+        RocksUtils.closeAll(db, options);
     }
 
-    /** {@inheritDoc} */
     @Override
-    public CompletableFuture<VaultEntry> get(ByteArray key) {
-        return supplyAsync(() -> {
-            try {
-                byte[] value = db.get(key.bytes());
-
-                return value == null ? null : new VaultEntry(key, value);
-            } catch (RocksDBException e) {
-                throw new IgniteInternalException("Unable to read data from RocksDB", e);
-            }
-        });
+    public VaultEntry get(ByteArray key) {
```

Comment:
```suggestion
    public @Nullable VaultEntry get(ByteArray key) {
```

Reply:
This is not needed, this annotation is inherited (at least IDEA doesn't complain)

Reply:
In my opinion, it is better to leave consistency in the signature, so that when viewing the implementation, there is no confusion that the abstraction can return `null`, but here at first glance never.